v2/client/client.go

```diff
@@ -47,6 +47,11 @@ type Client interface {
 	// * func(event.Event) (*event.Event, protocol.Result)
 	// * func(context.Context, event.Event) *event.Event
 	// * func(context.Context, event.Event) (*event.Event, protocol.Result)
+	// Details about the "fn" function returned values:
```

Comment:
IMHO we don't need to go into too much detail here. What do you think about changing all the `protocol.Result` returns in this doc string to `error` to keep it simple?

Reply:
Done